websites/utils.py

```diff
@@ -40,3 +40,13 @@ def set_dict_field(obj: Dict, field_path: str, value: Any):
 def get_dict_query_field(dict_field_name: str, sub_field: str):
     """Generate django query key for searching a nested json feild"""
     return dict_field_name + "__" + sub_field.replace(".", "__")
+
+
+def get_valid_base_filename(filename: str, content_type: str) -> str:
+    """Avoid forbidden filenames that could confuse hugo"""
+    if (
+        content_type in (constants.CONTENT_TYPE_PAGE, constants.CONTENT_TYPE_RESOURCE)
+        and filename in constants.CONTENT_FILENAMES_FORBIDDEN
```

Comment:
is it possible we want to do this for other content types too?

Reply:
@gumaerc ?

Reply:
It should cover anything that goes into the `content` folder.  Right now, "Pages" and "Resources" cover all bases but in the future if we add additional types of content this will have to cover that.  In addition, in the future if we add explicit support for nested content (`folders with `_index.md` and child content) or [Page Bundles](https://gohugo.io/content-management/page-bundles/) (folders with `index.md` and static content alongside it) then we will want to intentionally publish these files.  However, there's no scenario where user-generated content should place these files unintentionally.

Reply:
We also have other types of content in the ocw-www course where I imagine we'd also want this protection